Non-latin symbols transform into underlines

ndrwrk opened the pull request:
When uploading files via "Documents" or "Wiki", non-latin symbols transform into underlines
![screenshot-15](https://cloud.githubusercontent.com/assets/18718948/14884685/268bad2a-0d4f-11e6-8f54-eb5c2533594d.jpg)
![screenshot-18](https://cloud.githubusercontent.com/assets/18718948/14884686/268cbe22-0d4f-11e6-89a6-cd1d25141276.jpg)

oliverguenther committed:
Collapse main menu early on loading the DOM

This really is a hack around the initialization of angular for the main
menu. Since the state of the menu is determined through sessionStorage,
we can leverage the immediate DOM placement to modify the menu before
the full page has been loaded.

Otherwise, the main menu flickers when collapsed (i.e., is first loaded
expanded and when the page is loaded, is being collapsed).

oliverguenther committed:
Disable workaround in test mode

furinvader committed:
Remove redundant source property from HalResource

furinvader committed:
Add apiV3 service tests

furinvader committed:
Remove Restangular methods from HalResource $source

furinvader committed:
Add tests for the _plain property of resources

furinvader committed:
Make $source public

furinvader committed:
Refactor the HalResource a little

HDinger committed:
Fix bug with column width and avoid overflow of text

HDinger committed:
Add line seperator for dropdown menus in sidebar

furinvader committed:
Make properties of $links and $embedded non-settable

furinvader committed:
Return empty object if the resource's source does not provide _links or _embedded

furinvader committed:
Proxy the properties of the object to the source

furinvader committed:
Refactor $links

furinvader committed:
Refactor $embedded

furinvader committed:
Implement test for the href property

oliverguenther committed:
Add duration field

ulferts committed:
fix class selector determining when to hide the date picker

Without it, the date picker would be closed when the user tries to select a date

oliverguenther committed:
Add integration test

oliverguenther committed:
Merge pull request #4232 from ulferts/fix/inline_edit_date_picker

fix class selector determining when to hide the date picker

oliverguenther committed:
Merge pull request #4231 from HDinger/feature/22554-inline-edit-date-picker

[22554] Styling inplace edit date picker

ulferts committed:
Merge pull request #4218 from oliverguenther/feature/cost-object

[22559] Allow plugins to receive field classes

ulferts committed:
make required parameter explicit

furinvader committed:
Make property links lazy

ulferts committed:
initial request spec for editing project on wp resource

ulferts committed:
add project to payload representer

ulferts committed:
add specs for other linked attributes on wp payload resource

HDinger committed:
Introduce -small class for smaller inplace editing fields

ulferts committed:
Merge pull request #4230 from oliverguenther/feature/inline-edit-duration

Add duration field

furinvader committed:
Make the links lazy

ulferts committed:
define project changing behavior in wp contract

oliverguenther committed:
Separate wp-list controller query loading into services

oliverguenther committed:
Fix integration test after correct blur'ing

furinvader committed:
Make the embedded properties lazy

oliverguenther committed:
Remove double TS import

ulferts committed:
use variable in shared context

furinvader committed:
Make proxied properties enumerable

HDinger committed:
Insert separator even without hover or selection

furinvader committed:
Make the properties of the embedded elements lazy

furinvader committed:
Refactor embedded array element transformaiton

HDinger committed:
Introduce new color variable for the toggle separator

ulferts committed:
allow setting new custom field values when changing the project

oliverguenther committed:
Merge pull request #4227 from HDinger/feature/22845-add-line-to-separate-work-package-and-wiki-button

[22845] Add line to separate work package and wiki button

furinvader committed:
Refactor HalResource

furinvader committed:
Add make 'lazy' function a service

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Remove PaginationService setting from params

ulferts committed:
add block to controll when a notifiation is send

ulferts committed:
add capability to move work package accross projects to service

oliverguenther committed:
Load schema to determine writable state

oliverguenther committed:
Skip activation when already activated

ng-aria probably adds keypress handlers to ng-click handlers for
screenreaders, such as for space. This triggers `activate` of the inline
edit and causes the pristine copy to be invalid.

ulferts committed:
add service specific for adding notes to a work package

furinvader committed:
Rename hal sub folder to hal-resources

oliverguenther committed:
Very crudely reload table

oliverguenther committed:
Merge pull request #4225 from oliverguenther/feature/expand-error-columns

[22556] Error handling for other required/invalid fields of a WP row

oliverguenther committed:
Make escape cancel edition and restore pristine value

oliverguenther committed:
Copy live work packages based on lockVersion

oliverguenther committed:
Merge pull request #4233 from oliverguenther/feature/escape-cancels

Make escape cancel edition and restore pristine value

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Use restangular#withConfig to initialize uncached provider

oliverguenther committed:
Fix schema loading

oliverguenther committed:
Wrap $http for caching

We'll need an option to disable caching for specific routes, or ideally,
specify other means of caching requests.

This commit adds:
  - A wrapper for $http to catch caching options passed through
  Restangular
  - An improved CacheService that can handle loading promises
  (previously, it interfaced only to Resources)

HDinger committed:
Merge remote-tracking branch 'origin/release/5.0' into dev

HDinger committed:
Set tabindex to avoid mutliple tabs for one field

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4240 from HDinger/fix/19689-details-pane-relations-tab-focus-is-not-set-on-input-field-when-expanding-field

[19689] In details pane in relations tab, the focus is not set correctly on input field when expanding field

ulferts committed:
click on the only space left for selecting a row

ulferts committed:
be more generous when waiting for the loading indicator

oliverguenther committed:
Very basic animation for removed rows

ulferts committed:
reapply b52b88a

oliverguenther committed:
Disable cache for queries

ulferts committed:
remove unused method

oliverguenther committed:
Merge pull request #4241 from ulferts/fix/select_work_package_row_spec

Fix/select work package row spec

oliverguenther committed:
Cache resources in halResource

manuschiller committed:
[22752] show all fields in edit mode

ulferts committed:
add resource for querying the list of allowed projects

oliverguenther committed:
Remove uncachedProvider

We'll no longer need that one

oliverguenther committed:
Keep loading Promise in Hal resource

oliverguenther committed:
Add Internet Explorer 10 as unsupported

ulferts committed:
update documentation to reflect changes to wp's project

manuschiller committed:
[22624] Add link to success message on work package update/create

manuschiller committed:
[22624] Add link to success message on work package update/create

ulferts committed:
add Project as an editable field in the edit field config

ulferts committed:
move contracts from lib/api to app/contracts

oliverguenther committed:
Work package table: Add link to id column

ulferts committed:
replace under the hood expectations in feature spec by observable behaviour

oliverguenther committed:
Merge pull request #4247 from oliverguenther/feature/table-id-link

Work package table: Add link to id column

oliverguenther committed:
Remove feature-flags

We no longer disable detailsView, thus save the extra request.

oliverguenther committed:
Merge pull request #4246 from ulferts/feature/edit_work_packages_project

Feature/edit work packages project

oliverguenther committed:
Remove automatic opening of split pane

furinvader committed:
Replace _$loadingPromise with $self and make it public

furinvader committed:
Remove TODO comments

oliverguenther committed:
Add split view button to hovered rows

ulferts committed:
extract contract related methods of service into concern

ulferts committed:
remove offending is_*?

HDinger committed:
Change styling of split view icon in wp table

ulferts committed:
add scope to Type to query for those enabled in a project

HDinger committed:
Change cursor and shrink line height of wp inline edit fields

oliverguenther committed:
Merge pull request #4245 from oliverguenther/housekeeping/ie10-unsupported-browser

Add Internet Explorer 10 as unsupported

ulferts committed:
fix contract path in op_engine

ulferts committed:
improve CreateWorkPackageService to conform with new interface

ulferts committed:
Remove project#add_work_package

openprojectci committed:
Update reference to OpenProject-Translations

furinvader committed:
Move error resource service to hal-resource folder

furinvader committed:
Fix work package save spec

furinvader committed:
Merge pull request #4238 from furinvader/feature/hal-resource-performance

Tweak HalResource Performance

oliverguenther committed:
Merge pull request #4257 from ulferts/fix/contract_path_in_op_engine

fix contract path in op_engine

oliverguenther committed:
Merge pull request #4253 from oliverguenther/housekeeping/remove-feature-flags

Remove feature-flags

HDinger committed:
Set cursor for ID field

ulferts committed:
add a project independent endpoint for work package creation

ulferts committed:
move CreateFormResource to CreateProjectFormResource

oliverguenther committed:
Merge pull request #4254 from oliverguenther/feature/wp-table/split-view-button

Add split view button on hovered work package rows

oliverguenther committed:
Merge pull request #4255 from HDinger/feature/inline-edit-cursor-line-hight

[18404] Styling inline edit: Change cursor and line height

oliverguenther committed:
Add fullscreen link to table context menu

ulferts committed:
explicitly require to avoid "still in tree" error

ulferts committed:
Handle empty create requests

furinvader committed:
Linked resources are now settable using HalResources as values

oliverguenther committed:
Merge pull request #4239 from oliverguenther/feature/inline-edit/reload-table

[22563] Reload and Merge WP table in background

oliverguenther committed:
Merge pull request #4258 from furinvader/fix/linked-properties

Linked resources are now settable using HalResources as values

oliverguenther committed:
Fix duration parsing in inline-edit

ulferts committed:
add a project independent endpoint to create work packages

Includes the auxillary end points:
* form
* schema
* available_projects

oliverguenther committed:
Inline edit: Add tooltip when form not editable

This adds a toolbar to attributes which are writable from their schemas,
but are no longer writable when their form is loaded.

This may happen for, e.g., priority of parent work packages, but also
other magic constraints.

While we work on reducing the magic in the constraints, a generic tooltip
offers the fastest user-friendly way of informing the user of
constraints.

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4263 from oliverguenther/feature/inline-edit/22898-tooltip

[22898] Inline edit: Add tooltip when form not editable

oliverguenther committed:
Merge pull request #4261 from oliverguenther/fix/22829-duration-directive

[22829] Fix duration parsing in inline-edit

oliverguenther committed:
Remove private annotation for descendents of FormRepresenter

oliverguenther committed:
Set project and default type for project-scoped form

This was the default for new forms before and no type was set (e.g., creating a child)
and should not be broken.

HDinger committed:
Add additional condtition to avoid error message when clicking on ID

oliverguenther committed:
Specify new endpoints

Specifies the endpoints to be created by https://github.com/opf/openproject/pull/4262

[ci skip]

furinvader committed:
Remove parse error when opening details view

A little refactoring also took place to satisfy
the IDE - adding type definitions.
The distinction between arrow and ordinary
functions clarifies, where the arguments
object is needed and where it is not.

The useCaching variable always returned a
truthy value, because of the `|| true` at the
end. After removing it, the application startet
to function normally.

oliverguenther committed:
Merge pull request #4267 from oliverguenther/feature/independent-wp-doc

Specify new endpoints

oliverguenther committed:
Merge pull request #4262 from ulferts/feature/project_independent_work_package_create

Feature/project independent work package create

openprojectci committed:
Update reference to OpenProject-Translations

HDinger committed:
Merge pull request #4266 from HDinger/fix/avoid-error-message-when-clicking-on-id

[Inline Editing] Avoid error message when clicking on ID

oliverguenther committed:
Merge pull request #4268 from furinvader/fix/parse-error

Remove parse error when opening details view

oliverguenther committed:
Merge pull request #4249 from manuschiller/22752

[22752] show all fields in edit mode

HDinger committed:
Merge remote-tracking branch 'origin/release/5.0' into dev

oliverguenther committed:
Merge pull request #4252 from oliverguenther/feature/full-screen-context-item

Add fullscreen link to table context menu

HDinger committed:
Fix bug on wiki page

HDinger committed:
Remove underline from icon on hover

oliverguenther committed:
Merge pull request #4270 from HDinger/fix/inline-edit-icon-no-hover

[Inline Edit] Remove underline form split view icon in WP table

RobinWagner committed:
Change button text wiki

Change button text for wiki create button.

manuschiller committed:
Correct controlling when the link should be displayed and I18-Support

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4251 from manuschiller/22624

[22624] Add link to success message on work package update/create

oliverguenther committed:
Fix Wiki const lookup

oliverguenther committed:
Merge pull request #4274 from oliverguenther/change-wiki-button-text

Change button text wiki

oliverguenther committed:
Close notifications before creating new ones

oliverguenther committed:
Merge pull request #4275 from oliverguenther/fix/specs

Fix current broken specs

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge branch 'release/5.0' into dev

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

furinvader committed:
Fix karma tests

* Exit karma with the given exit code after tests have run

* Fix NotificationBox directive test

* Set karma singleRun to true

* Refactor loadingIndicator service

* Fix apiV3 service tests

* Fix HalLink service tests

* Fix HalResource tests

* Reformat WP list controller test code

* Fix WP list controller tests

* Refactor wpContextMenu and fix its tests

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
[22916] Inline edit only on attribute text, double click to fullscreen

* Reduce editable field to wp-table--cell-span

* Allow macs to CMD-click for multiselect

* Set double click to full screen

* Set focus expectation to anchor

* Fix flickering in inline expectation

* Update selector for double clicks

* Simplify innerMostSpan

* Add spec for double click full view

manuschiller committed:
[20140] modals: fixes for accessibility of modals by keyboard

oliverguenther committed:
Avoid submitting an invalid query with empty status

oliverguenther committed:
Merge layouts into one

Use angular.html.erb as a nested layout which renders base itself.

oliverguenther committed:
Merge pull request #4211 from oliverguenther/fix/22704-navigation-flicker

[22704] Collapse main menu early on loading the DOM

oliverguenther committed:
[13702]  Default columns: Move subject left, remove priority

This moves the subject as a default column directly next to Type,
and removes prirority from the default set of columns.

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge branch 'release/5.0' into dev

openprojectci committed:
Update reference to OpenProject-Translations

HDinger committed:
Change field widths (#4312)

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Pass the decorated work package to CreateService (#4317)

Since associations will not be marked `dirty` when they're written by
the decorator, we are missing custom field values when creating new work
packages.

This reverts some of the changes introduced by the create service and
passes the decorated work package directly.

oliverguenther committed:
Merge branch 'release/5.0' into dev

HDinger committed:
Set html titles for wiki and wp pages (#4311)

Closely related to #4308 this PR sets the html titles for wiki and WP pages.

oliverguenther committed:
Avoid RuntimeError when aggregating with nil representers (#4321)

When aggregating columns, e.g., a custom field that contains nil values,
the AggregationGroup receives an empty represented.

This commit fixes that by allowing the represented object to be nil

```
RuntimeError (no represented object passed):
  lib/api/decorators/single.rb:54:in `initialize'
  lib/api/decorators/aggregation_group.rb:39:in `initialize'
  lib/api/v3/work_packages/work_package_list_helpers.rb:138:in `new'
  lib/api/v3/work_packages/work_package_list_helpers.rb:138:in `block in generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `each'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `map'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:127:in `apply_and_generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:42:in `work_packages_by_params'
  lib/api/v3/work_packages/work_packages_by_project_api.rb:42:in `block (2 levels) in <class:WorkPackagesByProjectAPI>'
  app/middleware/params_parser_with_exclusion.rb:38:in `call'
```

oliverguenther committed:
Introduce header and Rails caching for schemas

This commit introduces cached project-scoped work packages schema.
Their caches are expired on a key based on

- the Project ID
- the Type ID
- the allowed custom fields for the given project

This commit only caches schema, not form endpoints.

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

crohr committed:
Move email settings out of the YAMl configuration file, and into the Settings page

crohr committed:
Migrate email settings in an after_initialize block

Otherwise the Setting model is not available.

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

HDinger committed:
Add pop up menu to help link in top menu (#4332)

This adds a pop up menu to the help icon in the top menu.

oliverguenther committed:
[22851] Table Inline Create (#4290)

* Move the WP create button to the WP buttons directory

* Move the keepTab service to the wp-panels directory

* Move angular module definitions into separate file

* Move the WP create button into a separate file

* Use TS in WP create button directive

* Refactor the WP create button controller

* Move the has-context-menu directive to the components directory

* Refactor has-context-menu item focus

* Create new button directive and place in table

* Export and use wpButton module

* Add title to form properties

* Add -new modifier to new work package rows

* Add WP inline create directive

* Reduce error widths to cell span

* Save or update work package conditionally

* Mark work package as new

* Implement the WP inline create button

* Style button correctly

* Decide when to activate the field for new WPs

* Remove padding on -new row

* Avoid animation on added -new rows

* Avoid writing title when passed from new workPackage

* Remove jqMigrate warning

* Avoid animations for newly added rows

* Add cancel button to inline create row

* Track new work packages by custom id

If we don't track new work packages, ng-animate will freak out due to
the track by expression. We thus track work packages by setting a custom id set from the
current date.

* Disable focus on errors outside accessibility mode

* Add inline create cancel context menu

* Conditionally set focus on fields

Set focus on the first active field

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Remove remainingFilterNames $filter (#4333)

This replaces the filter for remaining filter types, which is executed
many times for any digest and causes a significant performance impact
with a watch on the query and according changes when adding/removing a
filter.

oliverguenther committed:
[22969] Switch subject and type (#4334)

This changes position of column "subject" and "type"

oliverguenther committed:
Use default loading of table for details pane (#4335)

oliverguenther committed:
Select query filter keys correctly (#4337)

Fixes a regression from https://github.com/opf/openproject/pull/4333

furinvader committed:
Refactor config service (#4339)

crohr committed:
Setup a sane default for email delivery method in case no config file present

crohr committed:
Ignore email migration when running settings table does not exist

crohr committed:
Actually, don't reload the config either if the settings table does not exist

machisuji committed:
don't redirect to type overview after update

machisuji committed:
work package form configuration

openprojectci committed:
Update reference to OpenProject-Translations

crohr committed:
Revert to previous default for email_delivery_method.

Otherwise this breaks tests

oliverguenther committed:
Show inline-create in global work package list (#4340)

When no project context is available, this fetches the
`available_projects` endpoint and suggests using the first project for
creating a new row.

crohr committed:
Allow to select the way email sending is configured (either from config file, or settings)

furinvader committed:
Undo removal of isModuleEnabled method

furinvader committed:
Merge pull request #4344 from furinvader/fix/config-service

Undo removal of isModuleEnabled method

machisuji committed:
don't force unrequired fields

crohr committed:
Do not specify a default delivery method. Add SMTP domain

crohr committed:
Add tests for new methods to configure email settings

crohr committed:
Do not clear cache in tests, since it appears it breaks other tests

oliverguenther committed:
Adjust int settings spec accordingly

oliverguenther committed:
Fix 'with_settings' specs

machisuji committed:
put visibility info into wp schema and use that

jzahnd committed:
Add information for default admin user to quick start (#4347)

oliverguenther committed:
Add support for parent-scoped with_settings data

machisuji committed:
feature spec

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Remove protractor E2E tests (#4346)

In agreement with @furinvader, we have decided to remove protractor
testing from OpenProject.

Frontend tests are implemented in karma and test single units of work as
before.
Since the freshness of APIv3 JSON mocks are getting ouf hand and we have
covered almost all E2E protractor tests with Rails, we're going to
resort to Rails to write concise *actual* integration test with correct data.

machisuji committed:
integrate custom field toggle into visibility

machisuji committed:
comments

machisuji committed:
don't set default value as it doesn't work with MySQL

oliverguenther committed:
Move settings listener out of application.js

oliverguenther committed:
Fix rubocop issues

oliverguenther committed:
[Inline Edit / Create] Changes and bug fixes (#4336)

* Open work package on single click when already in split view

* Replace (new) with empty string for new IDs

* Replace trash icon with cancel icon

oliverguenther committed:
Merge pull request #4351 from oliverguenther/feature/web-based-email-settings

* Move email settings out of the YAMl configuration file, and into the Settings page

* Migrate email settings in an after_initialize block

Otherwise the Setting model is not available.

* Setup a sane default for email delivery method in case no config file present

* Ignore email migration when running settings table does not exist

* Actually, don't reload the config either if the settings table does not exist

* Revert to previous default for email_delivery_method.

* Allow to select the way email sending is configured (either from config file, or settings)

* Do not specify a default delivery method. Add SMTP domain

* Add tests for new methods to configure email settings

* Do not clear cache in tests, since it appears it breaks other tests

* Adjust int settings spec accordingly

* Fix 'with_settings' specs

* Add support for parent-scoped with_settings data

* Move settings listener out of application.js

machisuji committed:
updated specs to work with new behaviour

oliverguenther committed:
Rewrite removed frontend tests with feature specs (#4350)

This PR adds two feature specs:

Replacement for work-packages-modal-focus-spec.js
Replacement for parent-relations-handler-spec.js
The remainig tests removed with #4346 are already covered as follows:

activity-with-revisions-spec.js (covered by activity_comments_spec.rb)
details-pane-editable-spec.js (covered by edit_work_package_spec.rb)
details-pane-show-hide-accessibility-spec.js (obsolete)
details-pane-spec.js (covered by several feature specs)

oliverguenther committed:
Merge pull request #4349 from opf/feature/work_package_form_configuration

Feature/work package form configuration

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Add type#updated_on to cache key

oliverguenther committed:
Add redirect for /styleguide

Adds a development-only route for `/styleguide` as an alias to
`/assets/styleguide.html`.

oliverguenther committed:
Bump livingstyleguide

Avoids redcarpet-3.0.0 XSS
https://github.com/livingstyleguide/livingstyleguide/issues/161

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge branch 'release/5.0' into dev

oliverguenther committed:
Remove gulp with simpler npm tasks

This removes `gulp` and provides a simpler `karma` and `webpack`,
`webpack-watch` npm scripts.

For actual development, users will need the Rails stack anyway, so it
won't make sense to keep a separate set of dependencies up to date.

* Instead of `gulp webpack`, use `npm run webpack`.
* Instead of `gulp watch`, use `npm run webpack-watch`, which is *drastically* faster than gulp watch-polling.
* Instead of `gulp tests:karma` use `npm run karma`, or just `npm test`.

This also removes all traces of express and restores the Rails
styleguide. It is not whitelisted for precompilation and is only
available in the context of the development mode.

To access the styleguide, use `/assets/styleguide.html`.

oliverguenther committed:
Use Proc to infer available type attribute visibility

Otherwise, migrations will break when a plugin tries to register new
params.

oliverguenther committed:
Ensure configuration is loaded

oliverguenther committed:
Preselect values of new work package from filter values (#4355)

This allows to assign default values for new work packages from the
current query filter.

Due to the incompatibility in APIexperimental query filters APIv3
values, this link is made solely for the types assignees and types.

furinvader committed:
Merge pull request #4356 from oliverguenther/housekeeping/remove-gulp

Remove gulp and add tasks with simple npm tasks

machisuji committed:
Merge pull request #4322 from oliverguenther/feature/schema-caching

[Performance] Introduce header and Rails caching for APIv3

oliverguenther committed:
Acessor for work_package in specific schema

oliverguenther committed:
Fix webpack Typescript warnings (#4357)

Resolve TypeScript warnings output merged earlier in #4355

machisuji committed:
Merge pull request #4358 from oliverguenther/fix/create-wp-schema

Acessor for work_package in specific schema

furinvader committed:
Refactor WP filter related components (#4348)

* Implement WP filter button directive

* Tidy up the WP inline create button controller

* Reformat WP list route code

* Move Filter model to components

* Move filter constants to filters component directory

* Move query filters directive

* Move query filter directive

* Add filter container directive

Vote furinvader for less copy pasta!

* Refactor filter visibility code

* Move filters helper

* Move query model service

* Make the query service a part of the filters module

* Remove getFilterCount test from WP list controller

* Fix query model module import

* Fix karma tests

Importing Angular modules within the test files
is not possible yet.

* Move filter visibility to a dedicated service

* Get rid of the FiltersHelper service

oliverguenther committed:
Allow directory access for third party plugins

Otherwise, apache will reject accessing the asset when a plugin adds a
subdirectory.

openprojectci committed:
Update reference to OpenProject-Translations

furinvader committed:
Fix TypeScript karma tests (#4361)

oliverguenther committed:
Show table even when empty (#4362)

openprojectci committed:
Update reference to OpenProject-Translations

machisuji committed:
wp form config polish (#4364)

* allow hiding fields with defaults; consider name_source

* sort types by name; move projects config to bottom

* updated translations (default => active)

* no extra visibility behaviour for new WPs

* allow hiding even type; always show details header though

otherwise the 'show all attributes' button may vanish

* updated specs to include new info

oliverguenther committed:
[Bugfix] Work package grouping header (#4360)

* Refactor how table grouping is applied and refreshed

Currently, changes to groupBy causes the work package to be immediately
refreshed before the actual groups are collected from the backend.
This watch has been removed, since its unneeded.

The work package table is now tracked by the row id and the current
group, which causes the ng-repeat to re-evaluate when the group changes,
thus fixing the group header bug.

Additionally, we now carry the groupByColumn in the wpTableService,
where the other grouping values are derived as well.

* Allow nil aggregation group

This should fix https://community.openproject.com/work_packages/22816/activity

* Hide empty group header when groupName not defined

This avoids showing an empty header for inline create row.
Also forces the create button to disappear when the table is explictly
refreshed (e.g., when grouping on another attribute).

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://dev.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://dev.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
[22624] Display 'Open this work package in full screen' correctly (#4363)

* Display 'Open this work package in full screen' correctly

Currently, the 'Click here to open this work package in fullscreen
view.' is not implemented correctly:

1. It is not only shown for update and create, but also delete in the
table
2. It is not shown for inline create / edit
3. It is shown in full screen

This reimplements the functionality without requring deep knowledge in
NotificationsService.

* Move NotificationsService

* Move notification box directive

* Move DetailsTabOverviewController

oliverguenther commented:
Hey @ndrwrk, 

please use a [feature request in our community](https://community.openproject.com/projects/openproject/work_packages/create_new?type=1) for reporting an issue.

It is technically not a bug, I believe we should lift the filename restrictions so you can be more liberal with attached filenames.

Best,
Oliver
